src/Avalonia.Base/Input/GamepadEventArgs.cs

```diff
@@ -0,0 +1,277 @@
+        Button10,
+        /// <summary>
+        /// Right Analog Stick Depressed
+        /// </summary>
+        Button11,
+        /// <summary>
+        /// Dpad Up
+        /// </summary>
+        Button12,
+        /// <summary>
+        /// Dpad Down
+        /// </summary>
+        Button13,
+        /// <summary>
+        /// Dpad Left
+        /// </summary>
+        Button14,
+        /// <summary>
+        /// Dpad Right
+        /// </summary>
+        Button15,
+    }
+
+#pragma warning disable CA1823 // Avoid unused private fields
+#pragma warning disable CA1815 // should implement equals
+    // this struct is a glorified fixed-buffer and these "unused fields" will be used. 
+    public struct GamepadButtons
+    {
+        private ButtonState _0;
+        private ButtonState _1;
+        private ButtonState _2;
+        private ButtonState _3;
+        private ButtonState _4;
+        private ButtonState _5;
+        private ButtonState _6;
+        private ButtonState _7;
+        private ButtonState _8;
+        private ButtonState _9;
+        private ButtonState _10;
+        private ButtonState _11;
+        private ButtonState _12;
+        private ButtonState _13;
+        private ButtonState _14;
+        private ButtonState _15;
+
+        public ButtonState this[int index] 
+        {
+            get
+            {
+                if (index < 0)
+                    throw new IndexOutOfRangeException();
+                if (index > 15)
+                    throw new IndexOutOfRangeException();
+                // unsafe black magic because NS2.0 doesn't support MemoryMarshal.CreateSpan 
+                return Unsafe.Add(ref _0, index);
+            }
+            set
+            {
+                if (index < 0)
+                    throw new IndexOutOfRangeException();
+                if (index > 15)
+                    throw new IndexOutOfRangeException();
+                // unsafe black magic because NS2.0 doesn't support MemoryMarshal.CreateSpan 
+                Unsafe.Add(ref _0, index) = value;
+            }
+        }
+    }
+#pragma warning restore CA1823 // Avoid unused private fields
+#pragma warning restore CA1815 // Avoid unused private fields
+
+    public struct ButtonState : IEquatable<ButtonState>
+    {
+        public bool Pressed { get; set; }
+        public bool Touched { get; set; }
+        public bool JustPressed { get; set; }
+        public bool JustReleased { get; set; }
+        public double Value { get; set; }
+
+        public readonly override bool Equals(object? obj)
+        {
+            return obj is ButtonState state && Equals(state);
+        }
+
+        public readonly bool Equals(ButtonState state)
+        {
+            return Pressed == state.Pressed &&
+                   Touched == state.Touched &&
+                   JustPressed == state.JustPressed &&
+                   Value == state.Value;
+        }
+
+        public readonly override int GetHashCode()
```

Comment:
Make it a record struct?

Reply:
Great suggestion! Taken and implemented.